test/tests/functional/pbs_reservations.py

```diff
@@ -120,6 +120,35 @@ def test_degraded_standing_reservations(self):
         self.server.expect(RESV, a, id=rid, interval=1, offset=offset,
                            attrop=PTL_AND)
 
+    def test_duration_end_resv(self):
+        """
+        To test if reservations can be made by using
+        Duration and EndTime, making the server calculate
+        the StartTime.
+        """
+        # create reservation to end in 5 seconds and lasts 1.
+        a = {'Resource_List.select': '1:ncpus=1',
+             'reserve_end': int(time.time()) + 5,
+             'reserve_duration': 1}
+        R = Reservation(TEST_USER, attrs=a)
+        R.unset_attributes(['reserve_start'])
+        rid = self.server.submit(R)
+
+        a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, a, id=rid)
+
+        # create reservation to end in 5 seconds and lasts 10.
```

Comment:
Aren't these really two different test cases?  Maybe you should break them out into two.

Reply:
I've moved them to a different test suite, since this tests the server more than the scheduler.

Reply:
My point is that you are testing two different things in one test.  It's usually a good idea to break them into two different tests so you get better details after the test is run.  If one side succeeded and the other failed, you'll see that.

Reply:
I forgot to say when I moved them to a new test suite, I made them separate test cases.